packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -163,13 +163,15 @@ LipoExecutable() {
     fi
   done
 
-  # Merge desired architectures.
-  local merged="${executable}_merged"
-  lipo -output "${merged}" -create "${all_executables[@]}"
-
-  # Replace the original executable with the thinned one and clean up.
-  cp -f -- "${merged}" "${executable}" > /dev/null
-  rm -f -- "${merged}" "${all_executables[@]}"
+  if [[ ${#all_executables[@]} > 0 ]]; then
```

Comment:
Just reading this without reading the code above, don't thin if no architectures are found but it rather de facto means >1. Add a comment that it's 0 when non-fat.